lib/src/chart/radar_chart/radar_chart_data.dart

```diff
@@ -0,0 +1,267 @@
+import 'dart:ui';
+
+import 'package:equatable/equatable.dart';
+import 'package:fl_chart/fl_chart.dart';
+import 'package:fl_chart/src/utils/lerp.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+typedef GetTitleByIndexFunction = String Function(int index);
+
+class RadarChartData extends BaseChartData {
+  //ToDo(payam) : document data classes
+  RadarChartData({
+    this.dataSets = const [],
+    this.fillColor = Colors.transparent,
+    this.chartBorderData = const BorderSide(color: Colors.black, width: 2),
+    this.radarTouchData,
+    this.getTitle,
+    @required this.titleCount,
+    this.titleTextStyle = const TextStyle(color: Colors.black, fontSize: 12),
+    this.titlePositionPercentageOffset = 0.2,
+    @required this.tickCount,
+    this.ticksTextStyle = const TextStyle(color: Colors.black, fontSize: 9),
+    this.tickBorderData = const BorderSide(color: Colors.black, width: 2),
+    this.gridBorderData = const BorderSide(color: Colors.black, width: 2),
+    FlBorderData borderData,
+  })  : assert(dataSets != null, "the dataSets field can't be null"),
+        assert(titleCount != null && titleCount >= 2, "RadarChart need's more then 2 titles"),
+        assert(tickCount != null && tickCount >= 2, "RadarChart need's more then 2 ticks"),
+        assert(
+          titlePositionPercentageOffset >= 0 && titlePositionPercentageOffset <= 1,
+          'titlePositionPercentageOffset must be something between 0 and 1 ',
+        ),
+        assert(
+          dataSets.firstWhere(
+                (element) => element.dataEntries.length != titleCount,
+                orElse: () => null,
+              ) ==
+              null,
+          'dataSets value count must be equal to titleCount value',
+        ),
+        super(borderData: borderData, touchData: radarTouchData);
+
+  final List<RadarDataSet> dataSets;
```

Comment:
Please define these final variables on top of the class definition.

I know it is better to define them below the class definition (like what you did).

But keep in mind that we need to keep consistency, and follow the rule (even if it is wrong). After merging this PR, we can fix all data classes by moving fields definition to below the class definition in a separated PR and workflow.